src/sentry/options/manager.py

```diff
@@ -172,35 +171,40 @@ def delete(self, key):
 
         return self.store.delete(opt)
 
-    def _value_is_of_type(self, type, value):
-        # TODO(dcramer): replace with basic types
-        if type in (unicode, str):
-            type = basestring
-        return isinstance(value, type)
-
     def register(self, key, default=None, type=None, flags=DEFAULT_FLAGS,
                  ttl=DEFAULT_KEY_TTL, grace=DEFAULT_KEY_GRACE):
         assert key not in self.registry, 'Option already registered: %r' % key
+
+        # If our default is a callable, execute it to
+        # see what value is returns, so we can use that to derive the type
+        if callable(default):
+            default_value = default()
+        else:
+            default_value = default
+
         # Guess type based on the default value
         if type is None:
             # the default value would be equivilent to '' if no type / default
             # is specified and we assume unicode for safety
             if default is None:
-                type = unicode
-                default = u''
-            elif isinstance(default, basestring):
-                type = unicode
-            else:
-                type = _type(default)
+                default = default_value = u''
+            type = type_from_value(default_value)
```

Comment:
This was just for convenience since if you declare a default value... we can just derive the type based on that to avoid being redundant.

I thought it was worth a little better UX when defining options since we'll have a lot.